pyproject.toml

```diff
@@ -8,17 +8,33 @@ dependencies = [
     "fastapi[standard]>=0.115.12",
     "httpx>=0.28.1",
     "pydantic-settings>=2.8.1",
-    "python-jose>=3.4.0",
+    "python-jose>=3.4.0"
 ]
 
-[tool.pytest.ini_options]
-pythonpath=["."]
-testpaths=["tests"]
-
-[dependency-groups]
+[project.optional-dependencies]
 dev = [
-    "polyfactory>=2.21.0",
     "pytest>=8.3.5",
     "pytest-mock>=3.14.0",
     "pytest-cov>=4.1.0",
+    "ruff>=0.4.4",
+    "polyfactory>=2.21.0"
 ]
+
+[tool.pytest.ini_options]
+pythonpath = ["."]
+testpaths = ["tests"]
+addopts = "--cov=auth --cov=deploy --cov=routers --cov=schemas --cov-report=term --cov-report=xml"
```

Comment:
the `deploy` folder shouldn't be included here, the deployment code is basically a separate thing but included in this repo so we can do automatic deployment more easily